Automattic/jetpack
# webpack-config: Add fork-ts-checker-webpack-plugin option

## Proposed changes:

We've decided to prefer `@babel/preset-typescript` over `ts-loader` for dealing with TypeScript in the context of Webpack. For generating the definition files, we're generally going to use `fork-ts-checker-webpack-plugin` (although `tsc --emitDeclarationOnly` is also an option).

To make that as easy as possible for people to work with, let's make it an option in the shared webpack config.

Also, since the code was starting to get cluttered, I sorted all the "plugins" into alphabetical order to make it easier to find.

### Other information:


## Jetpack product discussion


pdWQjU-FO-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Nothing should change outside of webpack-config itself. Actually testing use of the plugin will come with the followup PR that changes things to use it.
